### ISSUE TITLE
The control information of the bounding box seems to be lost in your implementation.

### ISSUE THREAD

**Issue by fayewong666999** [Sonettoo/AeroGen#6]
The control information of the bounding box seems to be lost in your implementation.
In "./ldm/modules/diffusionmodules/openaimodel_bbox_v2.py":
class TimestepEmbedSequential(nn.Sequential, TimestepBlock):
    """
    A sequential module that passes timestep embeddings to the children that
    support it as an extra input.
    """

    def forward(self, x, emb, context=None, control=None, mask=None):
        for layer in self:
            if isinstance(layer, TimestepBlock):
                x = layer(x, emb)
            elif isinstance(layer, SpatialTransformer):
                x = layer(x, context, control=None, mask=None)
            else:
                x = layer(x)
        return x

"Control=None and mask=None", why? Won't this cause the information of the bounding box to be lost?

**Comment by Sonettoo** [comment]
Thanks, A serious mistake, I made a minor adjustment and verified the weights file to confirm it was trained with the correct model settings. It should work properly now.

**Comment by fayewong666999** [comment]
Should the weight files also be adjusted? I observed that many of the parameters in the weight file regarding the border encoding (condition_tokenizer) were not updated.

**Comment by fayewong666999** [comment]
Should the weight files also be adjusted? I observed that many of the parameters in the weight file regarding the bounding box encoding (condition_tokenizer) were not updated.

**Comment by Sonettoo** [comment]
Yes, I'll update a version later. But condition_tokenizer should be updated correctly?
Below:
    def configure_optimizers(self):
        trainable_names=[]
        model_name=[]
        lr = self.learning_rate
        params = list(self.condition_tokenizer.parameters())

**Comment by fayewong666999** [comment]
You add them to the updated parameters, but they are not backpropagated to get updated. I have observed that some of the parameters in the weights file that you give are initialized with all zeros and remain all zeros after training.

**Comment by fayewong666999** [comment]
self.null_text_feature = torch.nn.Parameter(torch.zeros([self.in_dim]))
self.null_position_feature = torch.nn.Parameter(torch.zeros([self.position_dim]))
In the weight file you gave, the parameter values of these two parameters are still 0, which is a serious error, and your paper is completely different from your statement.

**Comment by Sonettoo** [comment]
The encoder section is not our innovative work, but rather an extension along the lines of what has gone before, so if you have questions you can discuss them together!

**Comment by fayewong666999** [comment]
I mean, with your current code logic, the parameters of the border encoder can't get any valid updates because control=None, mask=None

**Comment by fayewong666999** [comment]
One of our previous work was also based on GLIGEN. In the official and training weights given by GLIGEN, these parameters (self.null_text_feature, self.null_position_feature) are not zero after training, thank you.

**Comment by Sonettoo** [comment]
OK, I see what you mean, the previous version of the code was used to test the results of removing dual cross-attention (DCA), only in the experimental setup of LMA, so the text entered at this point also contains the boundary information：
## _cond_txt=torch.cat((cond_txt,control),dim=1)_
 as shown in the following annotated section, leaving out the above section
```
**# cond_txt=torch.cat((cond_txt,control),dim=1)**
        eps = diffusion_model(x=x_noisy, timesteps=t, context=cond_txt, control=control, category_control=category_control, mask_control=mask_ control
                            )
``
So I fixed the bug at your prompt (set control ! =None) and re-uploaded the weights file, so it should update correctly?
```

**Comment by fayewong666999** [comment]
Ok, thank you for your patient recovery, we will continue to follow&cite your work, thank you.

**Comment by fayewong666999** [comment]
Ok, thank you for your patient reply. We will continue to follow&cite your work, thank you.

**Comment by Sonettoo** [comment]
Thank you again for your reminder.

**Comment by fayewong666999** [comment]
I hope my slightly offensive reply didn't make you feel uncomfortable.

**Comment by Sonettoo** [comment]
Thanks for checking in. I’m open to discussing things further if needed

**Comment by fayewong666999** [comment]
Please close my issue after fixing these problems, thank you.
